grace/base.py

```diff
@@ -102,7 +102,7 @@ def delaunay_edges_from_nodes(
     # add edge nodes
     if update_graph:
         edge_attrs = {
-            GraphAttrs.EDGE_PREDICTION: 0.0,
+            GraphAttrs.EDGE_PREDICTION: (0, np.array([0.9, 0.1])),
```

Comment:
Can you elaborate on this - why are you using a tuple here? I assume this is `(label, logits)`?

In general, I'm not a huge fan of this as it involves indexing, and that can lead to unclear code and errors later. I would suggest that we have either a dictionary or a custom type, e.g.:

```python
class EdgePrediction:
    label: int
    logits: npt.NDArray
```

Reply:
What happens if you have multiple labels, i.e. the length of the array is variable?

Reply:
I primarily sources from the template code in `opt.ipynb` where this was the logic. @crangelsmith kindly suggested a dictionary, too, with `str` keys like "predicted_label", "probab_TN", "probab_TP" which is what the optimiser needs.